src/WAV.jl

```diff
@@ -248,25 +248,40 @@ end
 
 ieee_float_container_type(nbits) = (nbits == 32 ? Float32 : (nbits == 64 ? Float64 : error("$nbits bits is not supported for WAVE_FORMAT_IEEE_FLOAT.")))
 
+## ignore skip for pipelines
+function Base.skip(::Base.Process, Uint64) end
+function Base.skip(::Base.ProcessChain, Uint64) end
+
+function read_raw_sample(io::IO, len::Integer)
+    ispipe = isa(io, Union{Base.Process, Base.ProcessChain})
+    if ispipe
+        return read(io)
```

Comment:
Is the idea to read as many bytes as provided by the pipe? What happens if the pipe reads more or less bytes than expected by the calling code?

Reply:
right now this implementation will simply read until the pipe closes (which is what I need), but if you want to respect a subrange specification, then I suppose `Base.skip` for pipes could actually read (and ignore) the number of bytes to be skipped, and the above `read(io)` could be limited to reading the amount of bytes requested.  

I haven't tried, but I can imagine wanting more data than the pipe delivers or closing the pipe early will raise some errors.  

One problem with respecting `len` in this call is that for the test case, `sox` just indicates a large value for `len` when writing to a pipe, so that is why I am ignoring `len` here. 

Perhaps reading _at most_ `len` bytes from the pipe handles this all correctly.

Reply:
I'm not familiar with the workings of `sox`, so I will take your word that this works as expected. 

Could you use multiple dispatch instead of the `isa` runtime check?

Reply:
Ah yes that would make sense indeed, thanks